src/net/virtual_connection.rs

```diff
@@ -75,40 +73,27 @@ impl VirtualConnection {
         time.duration_since(self.last_sent)
     }
 
-    /// This will create a heartbeat packet that is expected to be sent over the network
-    pub fn create_and_process_heartbeat(&mut self, time: Instant) -> OutgoingPacket<'static> {
-        self.last_sent = time;
-        self.congestion_handler
-            .process_outgoing(self.acknowledge_handler.local_sequence_num(), time);
-
-        OutgoingPacketBuilder::new(&[])
-            .with_default_header(
-                PacketType::Heartbeat,
-                DeliveryGuarantee::Unreliable,
-                OrderingGuarantee::None,
-            )
-            .build()
-    }
-
     /// This will pre-process the given buffer to be sent over the network.
     pub fn process_outgoing<'a>(
         &mut self,
-        payload: &'a [u8],
-        delivery_guarantee: DeliveryGuarantee,
-        ordering_guarantee: OrderingGuarantee,
+        packet: PacketInfo<'a>,
         last_item_identifier: Option<SequenceNumber>,
         time: Instant,
-    ) -> Result<Outgoing<'a>> {
-        match delivery_guarantee {
+    ) -> Result<OutgoingPackets<'a>> {
+        self.last_sent = time;
+        match packet.delivery {
             DeliveryGuarantee::Unreliable => {
-                if payload.len() <= self.config.receive_buffer_max_size {
-                    let mut builder = OutgoingPacketBuilder::new(payload).with_default_header(
-                        PacketType::Packet,
-                        delivery_guarantee,
-                        ordering_guarantee,
-                    );
+                if packet.payload.len() <= self.config.receive_buffer_max_size {
+                    if packet.packet_type == PacketType::Heartbeat {
+                        // TODO (bug?) is this really required here?
```

Comment:
It is not a bug, it is an unfinished feature. Like it does update the RTT, but we don't do something with RTT

Reply:
I will replace this comment with "TODO unfinished feature regarding RTT measurement." 
OK?

Reply:
Yea that is good to me